lib/handler/retry-handler.js

```diff
@@ -193,6 +194,19 @@ class RetryHandler {
       this.resume = null
 
       if (statusCode !== 206) {
+        // Abort when status code is not partial content
+        // and payload is already consumed because downstream
+        // will concatenate response from two request wrongly.
+        if (this.consumed) {
+          this.abort(
+            new RequestRetryError('unsupported retry when payload is consumed', statusCode, {
```

Comment:
The error message is not very actionable. I would rephrase to something like
```suggestion
            new RequestRetryError('server does not support the range header and the payload was partially consumed', statusCode, {
```